spec/inputs/gelf_spec.rb

```diff
@@ -146,4 +146,41 @@
       expect(event.timestamp.usec).to be_within(1000).of(123456)
     end
   end
+
+  context "when an invalid JSON is fed to the listener" do
+    subject { LogStash::Inputs::Gelf.new_event(message, "host") }
+    let(:message) { "Invalid JSON message" }
+
+    context "legacy JSON parser output" do
+      before(:each) do
+        class << LogStash::Inputs::Gelf; alias_method(:parse, :legacy_parse); end
```

Comment:
check the [json_lines specs](https://github.com/logstash-plugins/logstash-codec-json_lines/blob/51805af5d7b324f64bfb65ce35cf2848c804a16e/spec/codecs/json_lines_spec.rb#L190-L217) I actually stubbed the parse method instead of `respond_to?`